src/shadow-embed.js

```diff
@@ -58,16 +59,27 @@ export function createShadowRoot(hostElement) {
     return existingRoot;
   }
 
+  // Define our shadowRoot object to be returned
+  let shadowRoot;
+
   // Native support.
   const shadowDomSupported = getShadowDomSupportedVersion();
   if (shadowDomSupported == ShadowDomVersion.V1) {
-    return hostElement.attachShadow({mode: 'open'});
+    shadowRoot = hostElement.attachShadow({mode: 'open'});
   } else if (shadowDomSupported == ShadowDomVersion.V0) {
-    return hostElement.createShadowRoot();
+    shadowRoot = hostElement.createShadowRoot();
+  } else {
+    // Polyfill.
+    shadowRoot = createShadowRootPolyfill(hostElement);
   }
 
-  // Polyfill.
-  return createShadowRootPolyfill(hostElement);
+  // Ensure that Shadow CSS is supported, and add an id if not
+  if (!isShadowCssSupported()) {
+    const randomId = Math.floor(Math.random() * 10000);
+    shadowRoot.id = `i-amphtml-sd-${randomId}`;
```

Comment:
As discussed offline, this won't work. I think instead it should be:
```
const rootId = `i-amphtml-sd-${randomId}`;
if (!shadowDomSupported || !shadowCssSupported) {
  shadowRoot.host.classList.add(rootId);
}
```

And later, when we transform CSS:
```
if (!shadowDomSupported || !shadowCssSupported) {
  transform(..., '.${rootId}');
}
```

The trick here is using `.` instead of `#` - basically we'll switch to class-name from ID selector prefixes.